### ISSUE TITLE
Recognize that get_defined_vars() takes unused parameters

### ISSUE THREAD

**Issue by sybrew** [sirbrillig/phpcs-variable-analysis#91]
Code speaks a thousand words.

```php
function some_function( $var_1 ) { // VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable

    $var_2 = get_something(); // VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable

    do_something( get_defined_vars() ); // this should negate above warnings.
}
```

**Comment by sirbrillig** [comment]
Wow, I somehow never knew about `get_defined_vars()` before. Thanks for brining this up!

So you're suggesting that if `get_defined_vars()` is called in a scope, it should disable all unused variable warnings within that scope. That does make sense in the sense that all variables in that scope are being read just by calling that function. 

🤔 I'm trying to think of any risks with doing that. Do you have any real-world examples of that function being used that I could reference?

**Comment by sybrew** [comment]
Cheers 😄 

Yea, Automattic doesn't seem too fond of that function. I don't know why 🤷‍♂; I only found one (convoluted) instance in Jetpack.

And yes, that's exactly what I'm suggesting!

Aside from the example provided above, I use it when passing variadic parameters to my file inclusion handler<sup>*</sup>. 

Here's a reference: https://github.com/sybrew/the-seo-framework/blob/9638de251d856937b135a57dd5d2d663e81a39b8/inc/classes/profile.class.php#L79-L102

The placement of the error is a bit odd in that reference, but that's a common issue with phpcs.
![image](https://user-images.githubusercontent.com/7662236/60034082-1e643080-96aa-11e9-8aea-44fa2b5b93ef.png)

<sup>*</sup> file inclusions from an object method will also pass `$this`, and make the object accessible in even the private scope, but that's a story for another issue.
